opm/output/eclipse/WindowedArray.hpp

```diff
@@ -0,0 +1,177 @@
+/*
+  Copyright (c) 2018 Statoil ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_WINDOWED_VECTOR_HPP
+#define OPM_WINDOWED_VECTOR_HPP
+
+#include <cassert>
+#include <iterator>
+#include <type_traits>
+#include <vector>
+
+#include <boost/range/iterator_range.hpp>
+
+namespace Opm { namespace RestartIO { namespace Helpers {
+
+    template <typename T>
+    class WindowedArray
+    {
+    private:
+        using Vector = std::vector<T>;
```

Comment:
I'd prefer just `std::vector<T>` throughout; now I have to look up - "What was this `Vector` again?"